tidy3d/components/data/data_array.py

```diff
@@ -113,9 +114,17 @@ def assign_data_attrs(cls, val):
         return val
 
     def _interp_validator(self, field_name: str = None) -> None:
-        """Make sure we can interp()/sel() the data."""
-        # NOTE: this does not check every 'DataArray' by default. Instead, when required, this check can be
-        # called from a validator, as is the case with 'CustomMedium' and 'CustomFieldSource'.
+        """Make sure we can interp()/sel() the data.
+
+        NOTE
+        ----
+        This does not check every 'DataArray' by default. Instead, when required, this check can be
+        called from a validator, as is the case with 'CustomMedium' and 'CustomFieldSource'.
+        """
+        # skip this validator if currently tracing for autograd because
+        # self.values will be dtype('object') and not interpolatable
+        if isbox(self.values.ravel()[0]):
```

Comment:
Thanks @yaugenst-flex

My only concern is whether `.ravel()` will be slow, but I suppose interpolation already in the validator will be a bit slow too. There's also `self.values.flat[0]`, I'm not sure the difference but this might be a little faster as it returns an iterator rather than a new array.

Another option is that I think when there are tracers in the `values`, then the full traced ArrayBox will be in  `self.attrs[AUTOGRAD_KEY]`, so we could check `if AUTOGRAD_KEY in self.attrs:`, although it's probably less robust.

Reply:
another option could be to skip if `self.values.dtype == object`

Reply:
ok sounds good to me (re flat vs ravel vs dtype).